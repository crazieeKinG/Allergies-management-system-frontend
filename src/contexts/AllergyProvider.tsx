import { createContext, useState } from "react";
import AllergyInterface, {
    AllergyContextInterface,
} from "../interfaces/allergy.interfaces";

interface Props {
    children: React.ReactNode;
}

export const AllergyContext = createContext<
    AllergyContextInterface | undefined
>(undefined);

const AllergyProvider = ({ children }: Props) => {
    const [allergy, setAllergy] = useState<AllergyInterface[]>([]);

    return (
        <AllergyContext.Provider value={{ allergy, setAllergy }}>
            {children}
        </AllergyContext.Provider>
    );
};

export default AllergyProvider;
