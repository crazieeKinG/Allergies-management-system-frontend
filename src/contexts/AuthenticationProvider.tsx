import React, { createContext, useState } from "react";
import AuthenticationContextInterface, {
    AuthenticationContextDataInterface,
} from "../interfaces/authentication.interfaces";

interface Props {
    children: React.ReactNode;
}

export const AuthenticationContext = createContext<
    AuthenticationContextInterface | undefined
>(undefined);

const AuthenticationProvider = ({ children }: Props) => {
    const initialAuthenticationData: AuthenticationContextDataInterface = {
        username: "Saajan Shrestha",
        accessToken:
            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJiMDVlYWE3YS1iYThiLTQxYzItYmViZC00N2I0ODRmMTUwYmIiLCJpYXQiOjE2NjQ0MjI5Nzh9.QZZc2h3noQYWnvrTRDHjj_fvhKw3Ri-vApTz6E1KIuw",
    };
    const [authentication, setAuthentication] = useState(
        initialAuthenticationData
    );

    return (
        <AuthenticationContext.Provider
            value={{ authentication, setAuthentication }}
        >
            {children}
        </AuthenticationContext.Provider>
    );
};

export default AuthenticationProvider;
