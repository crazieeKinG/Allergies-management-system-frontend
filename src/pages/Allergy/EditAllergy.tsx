import { Divider, Typography } from "antd";
import AllergyForm from "../../components/Allergy/AllergyForm";
import AllergyInterface from "../../interfaces/allergy.interfaces";

const EditAllergy = () => {
    const data: AllergyInterface = {
        id: "asd",
        allergyName: "namr",
        referredName: "s",
        riskLevel: "Mild",
    };

    return (
        <div>
            <Typography.Title level={3}>Edit allergy</Typography.Title>

            <Divider />

            <AllergyForm initialValue={data} />
        </div>
    );
};

export default EditAllergy;
