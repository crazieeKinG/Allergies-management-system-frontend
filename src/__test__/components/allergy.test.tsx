import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AllergyProvider, {
    AllergyContext,
} from "../../contexts/AllergyProvider";
import AuthenticationProvider from "../../contexts/AuthenticationProvider";
import AllergyInterface from "../../interfaces/allergy.interfaces";
import EditAllergy from "../../pages/Allergy/EditAllergy";
import ListAllergy from "../../pages/Allergy/ListAllergy";
import NewAllergy from "../../pages/Allergy/NewAllergy";
import SingleAllergy from "../../pages/Allergy/SingleAllergy";
import EditSymptom from "../../pages/Allergy/Symptom/EditSymptom";
import "../mocks/matchMedia";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: () => {
        return { id: "id" };
    },
}));

describe("Allergy pages", () => {
    const allergyList = [
        {
            id: "id",
            allergyName: "allergy-name",
            description: "allergy-description",
            referredName: "allergy-referred-name",
            photoUrl: "allergy-photoUrl",
            riskLevel: "risk-level",
            symptoms: [],
        },
    ] as AllergyInterface[];

    beforeEach(() => {
        Date.now = jest.fn().mockReturnValue(new Date("2022-12-26").getTime());
    });

    it("should generate allergy list page", () => {
        const view = render(
            <AuthenticationProvider>
                <AllergyProvider>
                    <BrowserRouter>
                        <ListAllergy />
                    </BrowserRouter>
                </AllergyProvider>
            </AuthenticationProvider>
        );
        expect(view.asFragment()).toMatchSnapshot();
    });
    it("should generate new allergy page", () => {
        const view = render(
            <AuthenticationProvider>
                <AllergyProvider>
                    <BrowserRouter>
                        <NewAllergy />
                    </BrowserRouter>
                </AllergyProvider>
            </AuthenticationProvider>
        );
        expect(view.asFragment()).toMatchSnapshot();
    });
    it("should generate edit allergy page", () => {
        const view = render(
            <AuthenticationProvider>
                <AllergyContext.Provider
                    value={{
                        allergy: allergyList,
                        setAllergy: jest.fn(),
                    }}
                >
                    <BrowserRouter>
                        <EditAllergy />
                    </BrowserRouter>
                </AllergyContext.Provider>
            </AuthenticationProvider>
        );
        expect(view.asFragment()).toMatchSnapshot();
    });
    it("should generate single allergy page", () => {
        const view = render(
            <AuthenticationProvider>
                <AllergyContext.Provider
                    value={{
                        allergy: allergyList,
                        setAllergy: jest.fn(),
                    }}
                >
                    <BrowserRouter>
                        <SingleAllergy />
                    </BrowserRouter>
                </AllergyContext.Provider>
            </AuthenticationProvider>
        );
        expect(view.asFragment()).toMatchSnapshot();
    });
    it("should generate edit symptom page", () => {
        const view = render(
            <AuthenticationProvider>
                <AllergyProvider>
                    <BrowserRouter>
                        <EditSymptom />
                    </BrowserRouter>
                </AllergyProvider>
            </AuthenticationProvider>
        );
        expect(view.asFragment()).toMatchSnapshot();
    });
});
