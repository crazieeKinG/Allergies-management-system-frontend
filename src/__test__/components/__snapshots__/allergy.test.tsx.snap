// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allergy pages should generate allergy list page 1`] = `
<DocumentFragment>
  <div
    class="ant-row ant-row-space-between"
  >
    <div
      class="ant-col ant-col-16"
    >
      <h4
        class="ant-typography"
      >
        Allergy List
      </h4>
      <div
        class="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <div
        class="ant-list ant-list-vertical ant-list-lg ant-list-split ant-list-something-after-last-item"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-list-empty-text"
            >
              <div
                class="ant-empty ant-empty-normal"
              >
                <div
                  class="ant-empty-image"
                >
                  <svg
                    class="ant-empty-img-simple"
                    height="41"
                    viewBox="0 0 64 41"
                    width="64"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      transform="translate(0 1)"
                    >
                      <ellipse
                        class="ant-empty-img-simple-ellipse"
                        cx="32"
                        cy="33"
                        rx="32"
                        ry="7"
                      />
                      <g
                        class="ant-empty-img-simple-g"
                        fill-rule="nonzero"
                      >
                        <path
                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                        />
                        <path
                          class="ant-empty-img-simple-path"
                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                        />
                      </g>
                    </g>
                  </svg>
                </div>
                <div
                  class="ant-empty-description"
                >
                  No Data
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-list-pagination"
        >
          <ul
            class="ant-pagination"
            unselectable="unselectable"
          >
            <li
              aria-disabled="true"
              class="ant-pagination-prev ant-pagination-disabled"
              title="Previous Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="left"
                  class="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-disabled"
              tabindex="0"
              title="1"
            >
              <a
                rel="nofollow"
              >
                1
              </a>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-next ant-pagination-disabled"
              title="Next Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="right"
                  class="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <div
      class="ant-col ant-col-5"
    >
      <h5
        class="ant-typography"
      >
        Filters
      </h5>
      <span
        class="ant-input-group-wrapper ant-input-search"
      >
        <span
          class="ant-input-wrapper ant-input-group"
        >
          <input
            class="ant-input"
            placeholder="Search..."
            type="text"
            value=""
          />
          <span
            class="ant-input-group-addon"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-icon-only ant-input-search-button"
              type="button"
            >
              <span
                aria-label="search"
                class="anticon anticon-search"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Allergy pages should generate edit allergy page 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="ant-typography"
    >
      Edit allergy
    </h3>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <form
      class="ant-form ant-form-horizontal ant-form-hide-required-mark"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class="ant-form-item-required"
              for="allergyName"
              title="Name"
            >
              Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  id="allergyName"
                  placeholder="Allergy name"
                  type="text"
                  value="allergy-name"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class=""
              for="referredName"
              title="Referred Name"
            >
              Referred Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="referredName"
                  placeholder="Allergy secondary name"
                  type="text"
                  value="allergy-referred-name"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class=""
              for="photo"
              title="Allergy picture"
            >
              Allergy picture
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class=""
                >
                  <div
                    class="ant-upload ant-upload-select ant-upload-select-picture"
                  >
                    <span
                      class="ant-upload"
                      role="button"
                      tabindex="0"
                    >
                      <input
                        accept=""
                        id="photo"
                        style="display: none;"
                        type="file"
                      />
                      <button
                        class="ant-btn ant-btn-default"
                        type="button"
                      >
                        <span
                          aria-label="upload"
                          class="anticon anticon-upload"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="upload"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                        <span>
                          Upload
                        </span>
                      </button>
                    </span>
                  </div>
                  <div
                    class="ant-upload-list ant-upload-list-picture"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class=""
              for="description"
              title="Description"
            >
              Description
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-input-textarea ant-input-textarea-show-count ant-input-textarea-in-form-item"
                  data-count="19 / 1000"
                >
                  <textarea
                    class="ant-input"
                    id="description"
                    placeholder="Description of allergy"
                  >
                    allergy-description
                  </textarea>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class="ant-form-item-required"
              for="riskLevel"
              title="Risk Level"
            >
              Risk Level
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="riskLevel_list_0"
                        aria-autocomplete="list"
                        aria-controls="riskLevel_list"
                        aria-haspopup="listbox"
                        aria-owns="riskLevel_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="riskLevel"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="risk-level"
                    >
                      risk-level
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-offset-4 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        Update
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn ant-btn-primary ant-btn-dangerous"
                      type="button"
                    >
                      <span>
                        Delete
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Allergy pages should generate edit symptom page 1`] = `
<DocumentFragment>
  <div
    class="ant-skeleton"
  >
    <div
      class="ant-skeleton-content"
    >
      <h3
        class="ant-skeleton-title"
        style="width: 38%;"
      />
      <ul
        class="ant-skeleton-paragraph"
      >
        <li />
        <li />
        <li
          style="width: 61%;"
        />
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Allergy pages should generate new allergy page 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="ant-typography"
    >
      Add new allergy
    </h3>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <form
      class="ant-form ant-form-horizontal ant-form-hide-required-mark"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class="ant-form-item-required"
              for="allergyName"
              title="Name"
            >
              Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  id="allergyName"
                  placeholder="Allergy name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class=""
              for="referredName"
              title="Referred Name"
            >
              Referred Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="referredName"
                  placeholder="Allergy secondary name"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class=""
              for="photo"
              title="Allergy picture"
            >
              Allergy picture
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <span
                  class=""
                >
                  <div
                    class="ant-upload ant-upload-select ant-upload-select-picture"
                  >
                    <span
                      class="ant-upload"
                      role="button"
                      tabindex="0"
                    >
                      <input
                        accept=""
                        id="photo"
                        style="display: none;"
                        type="file"
                      />
                      <button
                        class="ant-btn ant-btn-default"
                        type="button"
                      >
                        <span
                          aria-label="upload"
                          class="anticon anticon-upload"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="upload"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                        <span>
                          Upload
                        </span>
                      </button>
                    </span>
                  </div>
                  <div
                    class="ant-upload-list ant-upload-list-picture"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class=""
              for="description"
              title="Description"
            >
              Description
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-input-textarea ant-input-textarea-show-count ant-input-textarea-in-form-item"
                  data-count="0 / 1000"
                >
                  <textarea
                    class="ant-input"
                    id="description"
                    placeholder="Description of allergy"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-4 ant-form-item-label ant-form-item-label-left"
          >
            <label
              class="ant-form-item-required"
              for="riskLevel"
              title="Risk Level"
            >
              Risk Level
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="riskLevel_list_0"
                        aria-autocomplete="list"
                        aria-controls="riskLevel_list"
                        aria-haspopup="listbox"
                        aria-owns="riskLevel_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="riskLevel"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select risk level
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-offset-4 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-dashed"
                  style="width: 60%;"
                  type="button"
                >
                  <span
                    aria-label="plus"
                    class="anticon anticon-plus"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="plus"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <defs>
                        <style />
                      </defs>
                      <path
                        d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                      />
                      <path
                        d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                      />
                    </svg>
                  </span>
                  <span>
                    Add Symptom
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-col-offset-4 ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        Save
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Allergy pages should generate single allergy page 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ant-row ant-row-middle"
    >
      <div
        class="ant-col"
      >
        <h2
          class="ant-typography"
        >
          allergy-name
        </h2>
      </div>
      <div
        class="ant-col"
      >
        <h4
          class="ant-typography ant-typography-secondary"
        >
            allergy-referred-name 
          <a
            href="/edit/id"
          >
            <span
              aria-label="edit"
              class="anticon anticon-edit"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="edit"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                />
              </svg>
            </span>
          </a>
        </h4>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <img
      alt="allergy-name"
      src="allergy-photoUrl"
      width="300"
    />
    <h5
      class="ant-typography"
    >
      Description
    </h5>
    <div
      class="ant-typography"
    >
      allergy-description
    </div>
    <h5
      class="ant-typography"
    >
      Symptoms  
      <a
        href="/edit/symptom/id"
      >
        <span
          aria-label="edit"
          class="anticon anticon-edit"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="edit"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
            />
          </svg>
        </span>
      </a>
    </h5>
    <div
      class="ant-list ant-list-split"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-list-empty-text"
          >
            <div
              class="ant-empty ant-empty-normal"
            >
              <div
                class="ant-empty-image"
              >
                <svg
                  class="ant-empty-img-simple"
                  height="41"
                  viewBox="0 0 64 41"
                  width="64"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    transform="translate(0 1)"
                  >
                    <ellipse
                      class="ant-empty-img-simple-ellipse"
                      cx="32"
                      cy="33"
                      rx="32"
                      ry="7"
                    />
                    <g
                      class="ant-empty-img-simple-g"
                      fill-rule="nonzero"
                    >
                      <path
                        d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                      />
                      <path
                        class="ant-empty-img-simple-path"
                        d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                      />
                    </g>
                  </g>
                </svg>
              </div>
              <div
                class="ant-empty-description"
              >
                No Data
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
